src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -459,9 +459,10 @@ private ImmutableArray<SourceMethodTypeParameterSymbol> MakeTypeParameters(Diagn
             return result.ToImmutableAndFree();
         }
 
-        public override ImmutableArray<TypeParameterConstraintClause> GetTypeParameterConstraintClauses(bool canIgnoreNullableContext)
+        public override ImmutableArray<TypeParameterConstraintClause> GetTypeParameterConstraintClauses(bool canUseLightweightTypeConstraintBinding)
         {
-            if (!_lazyTypeParameterConstraints.HasValue(canIgnoreNullableContext))
+            // We're not using lightweight type constraint binding for local function because the risk of cycles is minimal (no overrides)
```

Comment:
Man does this feel like a statement that's someone is going to look at in a few years and go "well that was wrong"

Reply:
I also prefer us not having this difference by comparison to other type parameters.

---
In reply to: [502597181](https://github.com/dotnet/roslyn/pull/48270#discussion_r502597181) [](ancestors = 502597181)